src/lib/operators/table_scan/column_vs_value_table_scan_impl.cpp

```diff
@@ -127,6 +133,175 @@ void ColumnVsValueTableScanImpl::_scan_dictionary_segment(const BaseDictionarySe
   });
 }
 
+void ColumnVsValueTableScanImpl::_scan_sorted_segment(const BaseSegment& segment, const ChunkID chunk_id,
+                                                      PosList& matches,
+                                                      const std::shared_ptr<const PosList>& position_filter) const {
+  resolve_data_and_segment_type(segment, [&](const auto type, const auto& typed_segment) {
+    if constexpr (std::is_same_v<decltype(typed_segment), const ReferenceSegment&>) {
+      Fail("Expected ReferenceSegments to be handled before calling this method");
+    } else {
+      // TODO(johannes-schneider): extract because of code duplication
+      // early outs for dictionary segments
+      // TODO(cmfcmf): This breaks some of our tests.
+      /*
+      if (const auto* dictionary_segment = dynamic_cast<const BaseDictionarySegment*>(&segment)) {
+        const auto search_value_id = _get_search_value_id(*dictionary_segment);
+        auto iterable = create_iterable_from_attribute_vector(*dictionary_segment);
+        if (_value_matches_all(*dictionary_segment, search_value_id)) {
+          iterable.with_iterators(position_filter, [&](auto begin, auto end) {
+            static const auto always_true = [](const auto&) { return true; };
+            _scan_with_iterators<true>(always_true, begin, end, chunk_id, matches);
+          });
+          return;
+        }
+        if (_value_matches_none(*dictionary_segment, search_value_id)) {
+          return;
+        }
+      }*/
+
+      Assert(segment.sort_order().value() == OrderByMode::AscendingNullsLast ||
```

Comment:
It should not happen but we should assert that there is actually a value.

Reply:
This is exactly the problem we run into by not taking the object oriented approach and having all kinds of modes as enums. We need these checks to make debugging easier, should someone add another OrderByMode.

And what happens if somebody adds another OrderByMode? We will fail at runtime. Then we will probably have to fix all parts in our program where we act upon the OrderByMode.

Reply:
We see your point, but think it's unlikely that someone adds a new `OrderByMode`, since they correspond to the order modes of the SQL standard.

> Then we will probably have to fix all parts in our program where we act upon the OrderByMode.

Additionally, all our changes have been moved to just one class, making refactoring pretty easy.